### PR TITLE
Changed C# sample to use .NET Core 3.0 and more lean and secure alpine as base image

### PR DESCRIPTION
**Pull request by meteatamel**
## Proposed Changes

- Changed C# sample to use more lean and secure alpine as base image

### CODE REVIEW

File: docs/serving/samples/hello-world/helloworld-csharp/Program.cs [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
     public class Program
     {
         public static void Main(string[] args)
         {
-            CreateWebHostBuilder(args).Build().Run();
+            CreateHostBuilder(args).Build().Run();
         }
 
-        public static IWebHostBuilder CreateWebHostBuilder(string[] args)
+        public static IHostBuilder CreateHostBuilder(string[] args)
```
[the remark this answers is not in the document]
Comment (meteatamel) [reply docs/serving/samples/hello-world/helloworld-csharp/Program.cs]: Yes, this was a change in the default web template of ASP.NET Core 3.

File: docs/serving/samples/hello-world/helloworld-csharp/Startup.cs [hunk 2]
```diff
@@ -15,18 +16,23 @@ public void ConfigureServices(IServiceCollection services)
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
```
[the remark this answers is not in the document]
Comment (meteatamel) [reply docs/serving/samples/hello-world/helloworld-csharp/Startup.cs]: Same here

**Comment by evankanderson** [comment]
/approve

(Though I haven't installed the tools to double-check, this looks reasonable. My kingdom for a cloud builder template so I don't need to do crazy local docker builds to test these things out.)

**Pull request merged**